Tests/AL-Go-Helper.Test.ps1

```diff
@@ -145,6 +151,91 @@ Describe "RunPipeline Action Tests" {
+
+        $withOrgSettings = ReadSettings -baseFolder $tempName -project 'Project' -repoName 'repo' -workflowName 'Workflow' -branchName 'dev' -userName 'user'
+        $withOrgSettings.property1 | Should -Be 'user1'
+        $withOrgSettings.property2 | Should -Be 'workflow2'
+        $withOrgSettings.property3 | Should -Be 'repo3'
+        $withOrgSettings.property4 | Should -Be 'branch4'
+        $withOrgSettings.property5 | Should -Be 'multi5'
+        $withOrgSettings.property6 | Should -Be 'user6'
+        $withOrgSettings.property7 | Should -Be 'orgsetting7'
+        $withOrgSettings.arr1 | Should -Be @("org3","repo1","repo2")
+
+        $ENV:ALGoRepoSettings = @{ "property3" = "reposetting3"; "property8" = "reposetting8" } | ConvertTo-Json -Depth 99
+
+        $withRepoSettings = ReadSettings -baseFolder $tempName -project 'Project' -repoName 'repo' -workflowName 'Workflow' -branchName 'dev' -userName 'user'
+        $withRepoSettings.property1 | Should -Be 'user1'
+        $withRepoSettings.property2 | Should -Be 'workflow2'
+        $withRepoSettings.property3 | Should -Be 'reposetting3'
+        $withRepoSettings.property4 | Should -Be 'branch4'
+        $withRepoSettings.property5 | Should -Be 'multi5'
+        $withRepoSettings.property6 | Should -Be 'user6'
+        $withRepoSettings.property7 | Should -Be 'orgsetting7'
+        $withRepoSettings.property8 | Should -Be 'reposetting8'
+
+        $conditionalSettings = [ordered]@{
+            "conditionalSettings" = @(
+                @{
+                    "branches" = @( 'branchx', 'branchy' )
+                    "settings" = @{ "property3" = "branchxy"; "property4" = "branchxy" }
+                }
+                @{
+                    "repositories" = @( 'repox', 'repoy' )
+                    "settings" = @{ "property3" = "repoxy"; "property4" = "repoxy" }
+                }
+                @{
+                    "projects" = @( 'projectx', 'projecty' )
+                    "settings" = @{ "property3" = "projectxy"; "property4" = "projectxy" }
+                }
+                @{
+                    "workflows" = @( 'workflowx', 'workflowy' )
+                    "settings" = @{ "property3" = "workflowxy"; "property4" = "workflowxy" }
+                }
+                @{
+                    "users" = @( 'userx', 'usery' )
+                    "settings" = @{ "property3" = "userxy"; "property4" = "userxy" }
+                }
+                @{
+                    "branches" = @( 'branchx', 'branchy' )
+                    "projects" = @( 'projectx','projecty' )
+                    "settings" = @{ "property3" = "bpxy"; "property4" = "bpxy" }
+                }
+            )
+        }
+        $ENV:ALGoRepoSettings = $conditionalSettings | ConvertTo-Json -Depth 99
+
+        $conditionalSettings = ReadSettings -baseFolder $tempName -project 'Project' -repoName 'repo' -workflowName 'Workflow' -branchName 'branchy' -userName 'user'

```

Comment:
Some comments to describe the tests would be really useful.

Reply:
comments added